app/controllers/users_controller.rb

```diff
@@ -0,0 +1,68 @@
+class UsersController < ApplicationController
+
+  before_action :authorize_admin_token,except: [:create]
+  before_action :set_user, only: [:update, :destroy, :accept_pending_request]
+
+    def index
+      if  @current_user.roles.exists?("role_type": 'Admin')
```

Comment:
why you need this condition?
'before_action :authorize_admin_token' itself should validate whether the logged in user has access to a particular api.